### ISSUE TITLE
[Windows] UI Virtualization broken in Listviews

### ISSUE THREAD

**Issue by Larhei** [dotnet/maui#24343]
### Description

I´m not an expert on this topic but from my experience in other xaml stacks this is looking fishy to me.

When using a ListView or Collectionview on Windows bound to a collection of 1000 items i observe the folowing.
When scrolling this itemscollections fast i end up with 3000+ instances of a contentview used in the Datatemplate.
In addition to 300+ instances of Microsoft.UI.Xaml.Controls.ListViewItem on the native side of things. 
Beside the fact that the Cachelength of the ItemsStackpanel is 4 and my ViewportHeight is 500 and each Item has a Height of 100.
There should be round about 8 Instances of ListViewItem from my understanding and maybe also less instances of my Datatemplate on the Maui Side. Over 500 MB for an App only showing two Lists with an empty View just looks wrong to me.

### Steps to Reproduce

1. Check out Sample on WIndows.
2. Run Sample on Windows.
3. Scroll the Listview a view times up and down. Watch the output while scrolling.
4. Scroll the Collectionview up and down. Watch the output while scrolling.
5. Take a few Snapshots in order to kick GC
6. Compare Baseline to Last Snapshot

### Link to public reproduction project repository

https://github.com/Larhei/Maui-Issues/tree/main/ListVirtualisation

### Version with bug

8.0.80 SR8

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

windows10.0.19041

### Did you find any workaround?

Nope

### Relevant log output

```shell
None
```

**Comment by Larhei** [comment]
Tested also without Hotreload and in Release.
Applying Border like in first similar issue did not solve the issue :-(

**Comment by MitchBomcanhao** [comment]
are you putting your listview inside a stack layout? somehow I recall reading somewhere that you shouldn't do that because it'd not contain the list controls and would force the listview to initialise everything, resulting in what you're complaining about.
use a grid instead of a stack layout.

**Comment by MitchBomcanhao** [comment]
asked copilot:
**_Yes, placing a ListView inside a StackLayout in Xamarin.Forms can indeed break virtualization. This is because the StackLayout measures its children with an infinite height, causing the ListView to load all its items at once instead of virtualizing them. This can lead to performance issues, especially with large data sets.

To maintain virtualization, you can use a Grid or AbsoluteLayout instead of a StackLayout_**

**Comment by Larhei** [comment]
@MitchBomcanhao 
Well i specify a Height to my ListViews and Collectionview. So the native side should know my the viewport height... But I will change my layout and see what i come up with.

**Comment by Larhei** [comment]
Changing to Grid did not do the trick
![image](https://github.com/user-attachments/assets/fca2a6ad-bd0b-4c22-9e89-e8844688ed99)
![image](https://github.com/user-attachments/assets/fa08e678-e055-4840-bd88-ce95e4cf2af3)
![image](https://github.com/user-attachments/assets/cbd99519-f397-40c5-ae03-0ec3430a095c)

Memory still over 300mb

**Comment by MitchBomcanhao** [comment]
yes, when taking snapshots I also see the number of objects increasing.
oddly enough, when adding Border around my control (which was just a label), it made the presentation noticeably slower, with viewcells taking much longer to draw.

**Comment by kevinxufei** [comment]
This issue has been verified using Visual Studio 17.12 Preview 1 (8.0.80 & 8.0.72). Can repro on windows platform with sample project.

**Comment by jfversluis** [comment]
ListView is deprecated for .NET 10, so we probably won't be working on this anymore. Please migrate to CollectionView and help us make that the solid approach for this. Thanks!
